.vsts-ci.yml

```diff
@@ -145,20 +144,59 @@ extends:
                 --prepareMachine
                 /p:RID=$(_RID)
               displayName: Build
+            - script: |
+                codesign --entitlements ./entitlements.plist --force --options runtime --sign - ./artifacts/layout/dotnet-core-uninstall/dotnet-core-uninstall
+              displayName: Add entitlements.plist
             - task: ArchiveFiles@2
               condition: and(eq(variables['system.pullrequest.isfork'], false), eq(variables['_BuildConfig'], 'Release'))
               inputs:
-                rootFolderOrFile: 'artifacts/layout/dotnet-core-uninstall/'
+                rootFolderOrFile: 'artifacts/layout/dotnet-core-uninstall/dotnet-core-uninstall'
                 includeRootFolder: false
-                archiveType: 'tar'
-                tarCompression: 'gz'
-                archiveFile: '$(Build.ArtifactStagingDirectory)/dotnet-core-uninstall.tar.gz'
                 replaceExistingArchive: true
+                archiveType: 'zip'
```

Comment:
Doing .zip instead of .tar.gz as `EsrpCodeSigning` supports signing .zip files, not .tar.gz

Reply:
We could also pack as zip, sign, unzip, and pack as .tar.gz but that seems like a lot of extra steps to me

Reply:
Yeah, even for pkg signing we used to have to wrap it inside a zip when submitting. For non-Windows users, tar.gz is preferred, but we can wait to see if there are strong objections to the zip

Reply:
I think zip files compress per file, while .tar.gz compresses all the files together, so I suspect that makes it difficult to sign the files (?)

Reply:
I'd think Chet would have a weigh-in on changing from `.tar.gz` to `.zip`. But what Jacques said is also an option is to put them into a .zip, sign, unpack, then repack as .tar.gz. (This would be a thing I'd just throw a couple lines of PowerShell to do.)